### ISSUE TITLE
Global poll visibility & change configuration options

### ISSUE THREAD

**Issue by javiexin** [Wolfsblvt/advancedpolls#40]
Suggestion: we could use just one drop-down or radio-button control to specify the poll visibility, and another control to specify vote change possibilities, greatly simplifying the posting options while being much more complete.

Visibility options:
- Public: results are always shown (new option)
- Default: results are shown after first vote, or when poll has ended
- Vote completed: results are shown after last vote, or when poll has ended (new option)
- Private: results are shown only when poll has ended (same as "hidden" today)

Vote change options:
- No change: you vote once and only once, nothing can be changed after
- Incremental voting: you may vote as many times as you want while you have pending votes to send (for polls with multiple options to choose, or scoring polls), but once you send a vote you cannot change it (same as "incremental" today)
- Change: you may change your vote as many times as you want while the poll is open

Two things that are "kind of" here as well are:
- No-vote: could be an option to visibility (another level?), or an ACP control to change "vote peeking" into no-vote (as it is now)
- Remove vote: could be an option to change votes (another level?), or an ACP control to add this to "Change votes"

We could also save in storage, by having a single DB field for each of these, rather than multiple binary fields.

What do you think?  I can implement this if it makes sense.

**Comment by Wolfsblvt** [comment]
Hmh... I know what you want to achieve with that, but I am not quite sure about that...
We shouldn't prevent use of options that might go well together, but we SHOULD prevent usage of options that will conflict.
Do we have some events that are conflicting?

If you think we really need this this could be okay. Otherwise I would say we should just keep the current setting. There is nothing bad about having many columns in database ore some options where you can choose from.

**Comment by javiexin** [comment]
The thing is that there are some options that only make sense on certain cases, not others.
For example, "incremental" votes only makes sense when changes are not allowed.  Likewise, no-vote only makes sense with no changes allowed.  Remove votes would be another option that only makes sense with vote changes.
On the other end, hidden votes and public votes are mutually exclusive options, and there are more (visibility) options in between.
I don't mind that much the storage issue, that would be just a plus.  But I think it makes sense, and it simplifies choices from the user point of view.

**Comment by Wolfsblvt** [comment]
Incremental and change votes are the only ones I can think of that are options where you have to choose.
What are "public polls" though?

I don't see how much options would conflict, so if this grouping is really needed.

**Comment by javiexin** [comment]
Public polls are polls where results are shown from the beginning, without any need for voting or anything else.  This is an option we do not have currently, but I am going to implement, as I need it to provide a mechanism to use polls for a kind of "demand recording", a place to record what, who and how much is needed of certain supplies.  These are directly the opposite to "hidden" polls, and definitely conflicting.

**Comment by Wolfsblvt** [comment]
Okay, Grouping would be okay for me, but only for directly interfering options, like the one you have mentioned. So if you are not sure if options would be working together, then may keep them seperated. Otherwise you can group them, if you want. I don't have a problem with that, if you can change all the code to match it and work with it (:

If you do that, don' worry about migrations though. You can change the existing (unreleased) ones instead of creating a new one. We should just keep the things that are already released. For us, we need to deinstall and install the extension again.

**Comment by Wolfsblvt** [comment]
Side note: I have given you access to this repository now, so that I can assign tasks to you and you may have more overview etc.
But **please** don't misuse that, okay? I don't want to have direct pushes from you to this repository. Only pull requests like before. Also don't change options or something that I have done :P

**Comment by javiexin** [comment]
Thanks a lot, I will NOT push anything, or change any config... don't worry!

**Comment by Wolfsblvt** [comment]
Great (: It will be now much clearer now with the open issues. I can assign them to me and to you, so we see wich belongs to whom.

Currently I am trying to set up the general testing environment and analysis tools and I will see how it goes.
After I have finished that stuff, you can can also take a look about the code reviewe by Scrutinizer and SensioInsideLabs for example. But mostly I will take that atm, no problem. Maybe I will create issues from them (in the future), for example to split up our poll medification method, etc, but we will see.

**Comment by javiexin** [comment]
Here is a mockup of the posting page with what I am thinking, but I do not want to continue until you take a look and give me your opinion.  I have tried to put enough relevant explanations in the page, but it might require more information.  You will have to tell me...  Here it goes:
![mockup_posting](https://cloud.githubusercontent.com/assets/11807050/7623651/9b600320-f9dd-11e4-9c89-c297ad153b47.png)

**Comment by Mauron** [comment]
Well, the concern about grouping poll options is understandable, since at the moment they take with this extensions more room than topic text field itself and I find it to be quite weird.
I'd choose **dropdown menus** anyway **with descriptions showed as tooltips** the way they appear in dropdown menus for user/group/forum permissions in ACP:
![Tooltips for drop-down menus](https://dl.dropboxusercontent.com/u/2103407/Cattura_924.png)
In particular, only when _Always_ or _First vote_  vote visibility options are selected, the _Preview results_ option should show up, possibly next to the drop-down instead of below (and the same should go for _No vote_ option).

As for issue #70, wouldn't it be better to **get it sorted out in phpBB 3.2** so that extensions (or phpBB itself) can **handle poll expiration basically as hours** (or even seconds, for conversion sake)?

**Comment by javiexin** [comment]
Thanks for the suggestion, it is a good one.
What do you think about completeness of options and need for the things that do not exist now?

As for 3.2 changes, that is beyond our capacity...
If you want, start a thread in Area51 and I will support you.

**Comment by Mauron** [comment]
Yes, I know the difference between phpBB developers and extensions ones. :smiley:
Anyway, apart from not having a profile on Area51, it would be difficult for me to explain on detail the issue with expire date for polls.

**Comment by Wolfsblvt** [comment]
Thanks for reporting this and for using the project. This project is no longer maintained, so I’m closing all remaining issues as part of archiving the repository. This avoids leaving a false impression that further development or support is planned.

Anyone interested in continuing development is welcome to do so through a fork, subject to the repository’s license and any applicable third-party licenses.

**Comment by Wolfsblvt** [comment]
Thanks for reporting this and for using the project. This repository is no longer maintained and is being archived, so I’m closing all remaining issues rather than leaving them open with the implication of future development or support. See the [Disclaimer](https://github.com/Wolfsblvt/advancedpolls#disclaimer) for details.
